### PR TITLE
Downstream mattermost-plugin-starter-template

### PR DESCRIPTION
**Pull request by hanzei**
This PR downstreams a lot of changes from https://github.com/mattermost/mattermost-plugin-starter-template namely:
- Start using Typescript
- Updates on Makefile
- Dependency updates

### CODE REVIEW

**Comment by hanzei** [comment]
@kaakaa Please check carefully for any issue

**Review by kaakaa — COMMENTED** [review]
Thanks @hanzei. I left a suggestion.

Other changes looks good to me, but almost all webapp code seems not to be formatted as .editorconfig. I'll work on it as another PR.

File: server/plugin/manifest.go [hunk 1]
```diff
@@ -2,10 +2,61 @@
...
+        "placeholder": "",
+        "default": false
+      }
+    ]
+  }
+}
+`
+
+func init() {
+	manifest = model.ManifestFromJson(strings.NewReader(manifestStr))
```
[the remark this answers is not in the document]
Comment (hanzei) [reply server/plugin/manifest.go:61]: Good suggestion :+1:

**Review by kaakaa — APPROVED** [review]
LGTM 👍

**Pull request merged**